draft-barnes-acme.md

```diff
@@ -218,304 +219,275 @@ In this section, we describe the four certificate management functions that ACME
 
-ACME uses a simple JSON-based structure for encoding signatures, based on the JSON Web Signature structure.  An ACME signature is a JSON object, with the following fields:
+| Code            | Semantic                                                 |
+|:================|:=========================================================|
+| malformed       | The request message was malformed                        |
+| unauthorized    | The client lacks sufficient authorization                |
+| serverInternal  | The server experienced an internal error                 |
+| badCSR          | The CSR is unacceptable (e.g., due to a short key)       |
 
-alg (required, string):
-: A token indicating the cryptographic algorithm used to compute the signature {{I-D.ietf-jose-json-web-algorithms}}. (MAC algorithms such as "HS*" MUST NOT be used.)
+Authorization and challenge objects can also contain error information to indicate why the server was unable to validate authorization.
 
-sig (required, string):
-: The signature, base64-encoded.
+TODO: Flesh out errors and syntax for them
 
-nonce (required, string):
-: A signer-provided random nonce of at least 16 bytes, base64-encoded.  (For anti-replay.)
+## Key Authorization
 
-jwk (required, object):
-: A JSON Web Key object describing the key used to verify the signature {{I-D.ietf-jose-json-web-key}}.
+The key authorization process establishes the authorization of an account key pair to manage certificates for a given identifier.  This process must assure the server of two things: First, that the client controls the private key of the key pair, and second, that the client holds the identifier in question.  This process may be repeated to associate multiple identifiers to a key pair (e.g., to request certificates with multiple identifiers), or to associate multiple key pairs with an identifier (e.g., for load balancing).
 
-Each usage of a signature object must specify the content being signed.  To avoid replay risk, the input to the signature algorithm is the concatenation of the nonce with the content to be signed.
+As illustrated by the figure in the overview section above, the authorization process proceeds in two phases.  The client first requests a new authorization, and then the server issues challenges that the client responds to.
 
-~~~~~~~~~~
-
-      signature-input = nonce || content
+To begin the key authorization process, the client sends a POST request to the server's new-authorization resource.  The body of the POST request MUST contain a JWS object, whose payload MUST be an initial authorization object.  This JWS object MUST contain the "identifier" field, so that the server knows what identifier is being authorized.  The client MAY provide contact information in the "contact" field in this or any subsequent request.
```

Comment:
Can you clarify here what an 'initial' authorization object means, versus a regular authorization object?